### ISSUE TITLE
Card can not be selected -> error message "Card unknown"

### ISSUE THREAD

**Issue by deepee2** [PersoSim/de.persosim.simulator#94]
System:
* Windows 10 20H2
* PersoSim_Driver_win_20180209 -> windows device manager contains the 2 smartcard readers "PersoSim Virtual Reader"
* openjdk version "1.8.0_282"

Steps to reproduce: 
* start PersoSim_0.17.0_win64
* Personalization -> Select predefined perso template -> Profile01.perso
* state "Card unknown" -> no usable card is found
![persosim_unknown-card](https://user-images.githubusercontent.com/11028366/108052401-5fae0e80-704c-11eb-97f7-d8505da2d691.png)
* logging contains no error, see attachment
[persosim_logging.txt](https://github.com/PersoSim/de.persosim.simulator/files/5987984/persosim_logging.txt)

**Comment by deepee2** [comment]
Seams to be caused by openjdk 8 (AdoptOpenJDK). 
I configured openjdk 11 (AdoptOpenJDK) and now card can be selected.

Please check that openjdk-issue and if persosim can not be used with openjdk 8 -> add a check and errormessage.

**Comment by deepee2** [comment]
openjdk 11 wasn't the solution. after a few days "Card unknown" is shown again. PersoSim can not be used anymore.

**Comment by dschulten** [comment]
In fact, the log does contain an error: 
```
2021-02-16T11:33:59.5 - DEBUG - this is not a valid public key
```
We see the problem only on specific machines. The log event is produced by the following code: 
https://github.com/PersoSim/de.persosim.simulator/blob/77607a64b43e7c29157994d459ac407f26309990/de.persosim.simulator/src/de/persosim/simulator/perso/xstream/KeyConverter.java#L105

```
try {
    pk = KeyFactory.getInstance(algorithmValue, Crypto.getCryptoProvider()).generatePublic(ks_pub);
} catch (InvalidKeySpecException| NoSuchAlgorithmException e1) {
    log(getClass(), "this is not a valid public key", LogLevel.DEBUG);
    // ...
}
```

Since the same .perso files work for several team members, the problem seems to be that the algorithms are not supported so that the public key gets rejected. I.e. I presume that NoSuchAlgorithmException gets thrown, not InvalidKeySpecException. And I suspect that the Java version decides what algorithms are supported.

The PersoSim Version is 0.17.2.20210224

I have several JDKs on my machine. Which Java Version will the PersoSim App use when executed in Explorer? The one in PATH or the one in %JAVA_HOME%? Is it possible to find out which Java executes PersoSim?

Would it be possible to log an error, not a debug event, and log the used algorithm and the exception message so that users can understand better what is going on?

**Comment by dschulten** [comment]
One can control via PersoSim.ini (same location as PersoSim.exe) which JDK runs PersoSim. Add the following at the end of the ini file:
```
-vm
<Path to your JDK bin folder, with forward slashes>
```

**Comment by cheindel** [comment]
Did you ever solve this?

**Comment by deepee2** [comment]
My solution was to disable 1 of the 2 smartcardreader named "PersoSim Virtual Reader" which are registered at windows device manager. I do not know if this was the cause but since then it works for me.

Windows 10 (21H2), openjdk 11.0.15, PersoSim 0.17.2.20210224

**Comment by cheindel** [comment]
Thank you @deepee2 ! That did the trick for us as well. We deleted all entries there and reinstalled the driver. Then it worked.

FAQ on secunet website also states this as a tip: "_Wichtig ist dabei, dass man ggf. zunächst die Reste der vorherigen Installationsversuche (=Virtuelle Kartenleser ohne Treiber) aus dem Geräte-Manager entfernt. Ansonsten sind anschließend mehrere virtuelle Leser installiert und PersoSim kommuniziert nicht korrekt mit dem Leser._"

So maybe this issue can be closed.

**Comment by AnReichle** [comment]
I tried to use a Profile created via PersoSim Editor and import it via "Personalization" -> "Select Perso from File". Subsequently I tries to access "my personal data" via AusweisApp.

Predefined Profiles seem to work, so it might be a problem within PersoSim Editor. But I get the same error in log as mentioned above, so I am not sure. Also in opposition to what is described above, PersoSim does visually indicate that the profile could be imported successfully (see screenshot), but access via AusweisApp to "my personal data" fails nonetheless, so PersoSim at least behaves inconsistently.

I tried it with several jdks and also checked, whether I have more than one Drivers PersoSim Virtual Reader registered - I have not.

PersoSim log ecxerpt (when importing custom perso from file):
```
2024-02-21T13:47:20.311+01:00[Europe/Berlin] - DEBUG - Select Perso Handler called with param: C:\Users\user\Desktop/bpurdy.perso
2024-02-21T13:47:20.311+01:00[Europe/Berlin] - DEBUG - executing command: loadperso C:\Users\user\Desktop/bpurdy.perso
2024-02-21T13:47:20.311+01:00[Europe/Berlin] - INFO  - Trying to load personalization for identifier "C:\Users\user\Desktop/bpurdy.perso"
2024-02-21T13:47:20.314+01:00[Europe/Berlin] - DEBUG - Available xstream converter services: none
2024-02-21T13:47:20.321+01:00[Europe/Berlin] - DEBUG - this is not a valid public key
2024-02-21T13:47:20.323+01:00[Europe/Berlin] - DEBUG - this is not a valid public key
2024-02-21T13:47:20.324+01:00[Europe/Berlin] - TRACE - init called
2024-02-21T13:47:20.324+01:00[Europe/Berlin] - TRACE - init finished
2024-02-21T13:47:23.352+01:00[Europe/Berlin] - INFO  - finished setting of personalization
```

AusweisApp log excerpt (when accessing personal information):
`2024.02.21 11:37:40.550 16108 C CardInfoFactory::readEfCardAccess(card/base/CardInfoFactory.cpp:129)       : Error while reading EF.CardAccess: Cannot read EF.CardAccess
`

Screenshot of state of PersoSim (after importing perso from file):
![PersoSim](https://github.com/PersoSim/de.persosim.simulator/assets/45435558/e29ce247-a901-4b1f-9ca2-32ab2c73f833)

System:

PersoSim 0.18.3.20220209
Windows 10 E
PersoSim_Driver_win_20180209
openlogic-openjdk-11.0.22+7-windows-x64

(change suffix to .perso)
[bpurdy.txt](https://github.com/PersoSim/de.persosim.simulator/files/14359847/bpurdy.txt)
